### DISCUSSION TITLE
`Tex.rebuild()` method

### DISCUSSION THREAD

**Discussion by kilacoda-old** [ManimCommunity/manim#2792] [Suggestions and Proposals]
## Description of proposed feature

A method, perhaps animatable which can be used to change an existing `Tex` mobject's LaTeX text and reinitialise it, creating a new mobject with differing content.

There have been queries regarding this on reddit and discord before, and the common reply we give is "create a new mobject and `Transform` to it", and apart from that, this might open up new possibilities for using LaTeX in manim, such as adding text at specific points in the compiled template, outside of the regularly placed one.

## How can the new feature be used?

I'm thinking something like this for a usecase:

```py
t = Tex("some long latex code")
t.append_in_environment("other text1") ## does not exist yet
t.append_before_environment("other text 2") ## this too
t.append_after_environment("other text 3") ## and this as well.

t = t.rebuild() ## similar to TexTemplate._rebuild()

self.add(t) ## displays the new mobject with the added text
```

**Comment by naveen521kk** [comment]
Or how about making `Tex` just a holder and the creation of the SVG files be done at the time of rendering. I mean like having an attribute where `text` can be changed and doesn't need a `.rebuild()` method.

It was previously discussed [here](https://discord.com/channels/581738731934056449/712140817200054294/846741612586860544) cc @Skaft.
